learning_resources/etl/utils.py

```diff
@@ -395,14 +332,50 @@ def documents_from_olx(
                 yield (
                     filebytes,
                     {
-                        "key": f"document_{next(counter)}_{filename}",
+                        "key": f"{path}/{filename}",
                         "content_type": CONTENT_TYPE_FILE,
                         "mime_type": mimetype,
                         "checksum": md5(filebytes).hexdigest(),  # noqa: S324
+                        "file_extension": extension_lower,
+                        "source_path": path,
                     },
                 )
 
 
+def clean_srt_content(content: str):
+    """
+    Remove timestamps and other extraneous data from SRT content
+
+    Args:
+        content (str): The SRT content to clean
+
+    Returns:
+        str: The content as a string without timestamps
+    """
+    # Remove timestamps
+    content = re.sub(
+        r"\d{2}:\d{2}:\d{2},\d{3} --> \d{2}:\d{2}:\d{2},\d{3}\n", "", content
+    )
+    # Remove sequence numbers
+    content = re.sub(r"\d+\n", "", content)
```

Comment:
it seems like there are a lot of srt files that still have timestamps - I think this is because of the '\n' at the end of the regex. I havent combed through all the cases that were missed but the ones i spotted ended with the timestamp which you should be able to fix by adjusting for end of match `r"\d{2}:\d{2}:\d{2},\d{3} --> \d{2}:\d{2}:\d{2},\d{3}(\n|$)"`

Reply:
also - should lock this down with some tests (or adjust existing tests)

Reply:
Thanks for the suggestion, I will make that change. Since for now we will just be using the content for vector search, a few stray timestamps are ok, we just don't want the entirety of the content to be interspersed by timestamps.